### PR TITLE
Single message drafts  

### PR DESCRIPTION
**Pull request by kaustubh-nair**
This PR adds support for a minimal draft message feature, storing a single
draft message when `meta s` is pressed.

### CODE REVIEW

File: zulipterminal/ui.py [hunk 1]
```diff
@@ -217,6 +217,22 @@ def keypress(self, size: Tuple[int, int], key: str) -> Optional[str]:
             search_box.set_edit_text("")
             self.controller.enter_editor_mode_with(search_box)
             return key
+        elif is_command_key('OPEN_DRAFT', key) and self.model.draft:
```
Comment (neiljp) [inline zulipterminal/ui.py:220]: If there is no draft, we should let the user know.

**Comment by neiljp** [comment]
@kaustubh-nair I just pushed the first commit as c7026b3b21ba1184d7f92a894b8877d661ceac6a with a reworded general commit text. I'm marking this as not needing review after our discussion on czo, but let me know if you want more details or discussion.

This should allow @preetmishra to implement #708 more simply too.

File: tests/ui_tools/test_boxes.py [hunk 2]
```diff
@@ -430,6 +430,8 @@ def test_keypress_typeahead_mode_autocomplete_key(self, mocker, write_box,
                                                       expect_footer_was_reset,
                                                       key):
         write_box.is_in_typeahead_mode = current_typeahead_mode
+        write_box.msg_write_box = mocker.Mock()
+        write_box.msg_write_box.edit_text = ''
```
Comment (neiljp) [inline tests/ui_tools/test_boxes.py:434]: Is this related?

File: zulipterminal/config/keys.py [hunk 3]
```diff
@@ -18,6 +18,16 @@
         'excluded_from_random_tips': True,
         'key_category': 'general',
     }),
+    ('SAVE_AS_DRAFT', {
+        'keys': {'meta s'},
+        'help_text': 'Save current message as a draft',
+        'key_category': 'general',
+    }),
```
Comment (neiljp) [inline zulipterminal/config/keys.py:25]: Shouldn't this be in the msg_compose category? Probably just below send?

File: zulipterminal/config/keys.py [hunk 4]
```diff
@@ -18,6 +18,16 @@
...
+    ('OPEN_DRAFT', {
+        'keys': {'d'},
+        'help_text': 'Open draft message saved in this session',
+        'key_category': 'general',
+    }),
```
Comment (neiljp) [inline zulipterminal/config/keys.py:30]: I'd suggest we want this below the about menu entry?

File: zulipterminal/ui_tools/boxes.py [hunk 5]
```diff
@@ -307,6 +307,26 @@ def keypress(self, size: urwid_Size, key: str) -> Optional[str]:
...
+                        display_recipient=self.stream_write_box.edit_text,
+                        content=self.msg_write_box.edit_text,
+                        subject=self.title_write_box.edit_text,
+                        stream_id=self.stream_id,
+                        type='stream',
+                    )
+                if not self.model.draft:
+                    self.model.save_draft(message)
+                if message != self.model.draft:
+                    self.view.controller.save_draft_confirmation_popup(message)
```
Comment (neiljp) [inline zulipterminal/ui_tools/boxes.py:329]: This is not updated to use the model method to access the draft.

File: zulipterminal/model.py [hunk 6]
```diff
@@ -285,6 +286,10 @@ def react_to_message(self,
             response = self.client.add_reaction(reaction_to_toggle_spec)
         display_error_if_present(response, self.controller)
 
+    def save_draft(self, message: Message) -> None:
+        self.draft = message
+        self.controller.view.set_footer_text("Saved message as draft", 3)
```
[the remark this answers is not in the document]
Comment (neiljp) [reply zulipterminal/model.py:294]: In the function that is called when 'yes' is selected.

File: zulipterminal/model.py [hunk 7]
```diff
@@ -285,6 +286,13 @@ def react_to_message(self,
             response = self.client.add_reaction(reaction_to_toggle_spec)
         display_error_if_present(response, self.controller)
 
+    def session_draft_message(self) -> Optional[Message]:
+        return self.draft
+
+    def save_draft(self, message: Message) -> None:
+        self.draft = message
```
Comment (neiljp) [inline zulipterminal/model.py:293]: I'm not sure why we're avoiding deepcopy here for encapsulation purposes.

File: zulipterminal/model.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (kaustubh-nair) [reply zulipterminal/model.py:294]: `model.save_draft()` _is_ the method that is called when 'yes' is selected? I'm confused :thinking:

File: tests/ui_tools/test_boxes.py (hunk 2 above)
Comment (kaustubh-nair) [reply tests/ui_tools/test_boxes.py:434]: Used in the older version I think - removed.

File: zulipterminal/model.py (hunk 7 above)
Comment (kaustubh-nair) [reply zulipterminal/model.py]: Fixed, also rename draft to _draft.

**Comment by kaustubh-nair** [comment]
@neiljp Thanks for the review, PR updated.

File: zulipterminal/model.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (neiljp) [reply zulipterminal/model.py:295]: `save_draft` is the method that's called currently, but I was wondering if we should call the `set_footer_text` from the popup too. We already have a lot of ui-coupled code in the model, which I'd like to move away from, but let's discuss and that can be something we do later.

**Comment by neiljp** [comment]
@kaustubh-nair This will be useful, thanks for working on this! :+1: I just merged this with a slight adjustment to the README to match the order of the hotkeys in the help menu :tada:
